File: module_wan/src/main/java/com/huyingbao/module/wan/ui/friend/store/FriendStore.java
package com.huyingbao.module.wan.ui.friend.store;

import com.huyingbao.core.arch.dispatcher.RxDispatcher;
import com.huyingbao.core.arch.model.RxAction;
import com.huyingbao.core.arch.store.RxFragmentStore;
import com.huyingbao.module.wan.action.WanResponse;
import com.huyingbao.module.wan.ui.friend.action.FriendAction;
import com.huyingbao.module.wan.ui.friend.model.WebSite;

import org.greenrobot.eventbus.Subscribe;

import java.util.ArrayList;

import javax.inject.Inject;
import javax.inject.Singleton;

import androidx.lifecycle.MutableLiveData;

/**
 * @author liujunfeng
 * @date 2019/1/1
 */
@Singleton
public class FriendStore extends RxFragmentStore {
    private final MutableLiveData<WanResponse<ArrayList<WebSite>>> mWebSiteListData = new MutableLiveData<>();
    private boolean mIsCreated;

    @Inject
    FriendStore(RxDispatcher rxDispatcher) {
        super(rxDispatcher);
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        mIsCreated = false;
        mWebSiteListData.setValue(null);
    }

    @Override
    @Subscribe
    public void onRxAction(RxAction rxAction) {
        switch (rxAction.getTag()) {
            case FriendAction.GET_FRIEND_LIST:
                mIsCreated = true;
                mWebSiteListData.setValue(rxAction.getResponse());
                break;
            default:
                break;
        }
    }

    public MutableLiveData<WanResponse<ArrayList<WebSite>>> getWebSiteListData() {
        return mWebSiteListData;
    }

    public boolean isCreated() {
        return mIsCreated;
    }
}

File: module_wan_kotlin/src/main/java/com/huyingbao/module/wan/ui/friend/store/FriendStore.kt
package com.huyingbao.module.wan.ui.friend.store

import androidx.lifecycle.MutableLiveData
import com.huyingbao.core.arch.dispatcher.RxDispatcher
import com.huyingbao.core.arch.model.RxAction
import com.huyingbao.core.arch.store.RxFragmentStore
import com.huyingbao.module.wan.action.WanResponse
import com.huyingbao.module.wan.ui.friend.action.FriendAction
import com.huyingbao.module.wan.ui.friend.model.WebSite
import org.greenrobot.eventbus.Subscribe
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by liujunfeng on 2019/1/1.
 */
@Singleton
class FriendStore @Inject
internal constructor(rxDispatcher: RxDispatcher) : RxFragmentStore(rxDispatcher) {
    val webSiteListData = MutableLiveData<WanResponse<ArrayList<WebSite>>>()
    var isCreated: Boolean = false
        private set

    override fun onCleared() {
        super.onCleared()
        isCreated = false
        webSiteListData.value = null
    }

    @Subscribe(tags = [FriendAction.GET_FRIEND_LIST])
    fun setWebSiteListData(rxAction: RxAction) {
        isCreated = true
        webSiteListData.value = rxAction.getResponse()
    }
}
